### ISSUE TITLE
no entries populating?

### ISSUE THREAD

**Issue by abathur** [SuprHackerSteve/Crescendo#18]
I suspect this is the same issue as #16, but I'm opening a new issue since I've never used Crescendo, can't swear I'm not just holding it wrong, and didn't want to spam the previous now-macless user with notifications. 

I did "allow" the system extension and approve the FDE access. The program appears to load clean, but nothing shows up after I click on Start.

I captured the following from Console (filtered to a search for `any:suprhackersteve`):
```
default	17:03:10.772943-0500	com.suprhackersteve.crescendo.CrescendoExtension	App disconnected.
default	17:03:10.773333-0500	com.suprhackersteve.crescendo.CrescendoExtension	Crescendo disabled.
default	17:03:13.565418-0500	distnoted	register name: com.apple.sharedfilelist.change object: com.apple.LSSharedFileList.ApplicationRecentDocuments/com.suprhackersteve.crescendo token: f49fe pid: 408
default	17:03:14.661448-0500	distnoted	register name: com.apple.sharedfilelist.change object: com.apple.LSSharedFileList.ApplicationRecentDocuments/com.suprhackersteve.crescendo token: f4a6c pid: 383
default	17:03:14.707180-0500	launchservicesd	CHECKIN:0x0-0x829829 23798 com.suprhackersteve.crescendo
default	17:03:14.771110-0500	distnoted	register name: com.apple.xctest.FakeForceTouchDevice object: com.suprhackersteve.crescendo token: f4268 pid: 23798
default	17:03:14.799288-0500	distnoted	register name: com.apple.nsquiet_safe_quit_give_reason object: com.suprhackersteve.crescendo token: f426e pid: 23798
default	17:03:15.062985-0500	tccd	-[TCCDAccessIdentity staticCode]: static code for: identifier com.suprhackersteve.crescendo, type: 0: 0x7f8e5d51b0d0 at /Applications/Crescendo.app
default	17:03:15.233119-0500	tccd	-[TCCDAccessIdentity staticCode]: static code for: identifier com.suprhackersteve.crescendo, type: 0: 0x7f8e5d40e120 at /Applications/Crescendo.app
default	17:03:16.477370-0500	sysextd	attempting to realize extension with identifier com.suprhackersteve.crescendo.CrescendoExtension
default	17:03:16.523751-0500	Crescendo	Trying to connect to service: AD94776VX5.com.suprhackersteve.crescendo.CrescendoExtension.xpc
default	17:03:16.525305-0500	com.suprhackersteve.crescendo.CrescendoExtension	App client connected.
error	17:03:16.527503-0500	com.suprhackersteve.crescendo.CrescendoExtension	Failed to open service: -536870174
default	17:03:16.527627-0500	com.suprhackersteve.crescendo.CrescendoExtension	Failed to initialize ES client: es_new_client_result_t(rawValue: 4)
default	17:03:16.527699-0500	com.suprhackersteve.crescendo.CrescendoExtension	Enabled Crescendo subsystem.
default	17:03:16.527736-0500	com.suprhackersteve.crescendo.CrescendoExtension	Created esclient.
error	17:03:16.610629-0500	sandboxd	Sandbox: com.suprhackerst(23096) System Policy: deny(1) system-privilege 1016
Violation:       System Policy: deny(1) system-privilege 1016 
Process:         com.suprhackerst [23096]
Path:            /Library/SystemExtensions/AFA5F0E5-7297-456F-A988-1DE006192220/com.suprhackersteve.crescendo.CrescendoExtension.systemextension/Contents/MacOS/com.suprhackersteve.crescendo.CrescendoExtension
Load Address:    0x10a68d000
Identifier:      com.suprhackersteve.crescendo.CrescendoExtension
Version:         35 (1.0.4)
Code Type:       x86_64 (Native)
Parent Process:  launchd [1]
Responsible:     /Library/SystemExtensions/AFA5F0E5-7297-456F-A988-1DE006192220/com.suprhackersteve.crescendo.CrescendoExtension.systemextension/Contents/MacOS/com.suprhackersteve.crescendo.CrescendoExtension
User ID:         0

Date/Time:       2020-08-02 17:03:16.554 CDT
OS Version:      Mac OS X 10.15.5 (19F101)
Report Version:  8


MetaData: {"primary-filter":"privilege-id","target":1016,"profile-flags":0,"apple-internal":false,"platform-policy":true,"team-id":"AD94776VX5","pid":23096,"flags":1029,"platform-binary":false,"user-approval":"kTCCServiceSystemPolicyAllFiles","build":"Mac OS X 10.15.5 (19F101)","process":"com.suprhackerst","profile":"platform","responsible-process-user-uuid":"FFFFEEEE-DDDD-CCCC-BBBB-AAAA00000000","hardware":"Mac","operation":"system-privilege","uid":0,"responsible-process-path":"\/Library\/SystemExtensions\/AFA5F0E5-7297-456F-A988-1DE006192220\/com.suprhackersteve.crescendo.CrescendoExtension.systemextension\/Contents\/MacOS\/com.suprhackersteve.crescendo.CrescendoExtension","errno":1,"process-path":"\/Library\/SystemExtensions\/AFA5F0E5-7297-456F-A988-1DE006192220\/com.suprhackersteve.crescendo.CrescendoExtension.systemextension\/Contents\/MacOS\/com.suprhackersteve.crescendo.CrescendoExtension","primary-filter-value":1016,"signing-id":"com.suprhackersteve.crescendo.CrescendoExtension","action":"deny","privilege-id":"PRIV_ENDPOINTSECURITY_CLIENT","summary":"deny(1) system-privilege 1016","platform_binary":"no","responsible-process-uid":0}

Thread 0 (id: 3453705):
0   libsystem_kernel.dylib        	0x00007fff6d107502 __sigsuspend_nocancel + 10
1   libdispatch.dylib             	0x00007fff6cf77476 _dispatch_sigsuspend + 0

Thread 1 (id: 3463850):
0   libsystem_kernel.dylib        	0x00007fff6d1034ce __workq_kernreturn + 10
1   libsystem_pthread.dylib       	0x00007fff6d1c0b77 start_wqthread + 15

Thread 2 (id: 3464271):
0   libsystem_kernel.dylib        	0x00007fff6d101dfa mach_msg_trap + 10
1   IOKit                         	0x00007fff35ed1472 io_service_open_extended + 137
2   IOKit                         	0x00007fff35ed13d8 IOServiceOpen + 39
3   libEndpointSecurity.dylib     	0x00007fff69ad4b6e es_new_client + 416
4   com.suprhackersteve.crescendo.CrescendoExtension	0x000000010a695e78
5   com.suprhackersteve.crescendo.CrescendoExtension	0x000000010a697e01
6   com.suprhackersteve.crescendo.CrescendoExtension	0x000000010a6960fe
7   libdispatch.dylib             	0x00007fff6cf67658 _dispatch_client_callout + 8
8   libdispatch.dylib             	0x00007fff6cf736ec _dispatch_lane_barrier_sync_invoke_and_complete + 60
9   com.suprhackersteve.crescendo.CrescendoExtension	0x000000010a695d82
10  com.suprhackersteve.crescendo.CrescendoExtension	0x000000010a698147
11  com.suprhackersteve.crescendo.CrescendoExtension	0x000000010a68ffc6
12  com.suprhackersteve.crescendo.CrescendoExtension	0x000000010a68f9d7
13  Foundation                    	0x00007fff35875413 __NSXPCCONNECTION_IS_CALLING_OUT_TO_EXPORTED_OBJECT_S1__ + 10
14  Foundation                    	0x00007fff357ff8de -[NSXPCConnection _decodeAndInvokeMessageWithEvent:flags:] + 2363
15  Foundation                    	0x00007fff357b6a49 message_handler + 210
16  libxpc.dylib                  	0x00007fff6d2052bc _xpc_connection_call_event_handler + 56
17  libxpc.dylib                  	0x00007fff6d2041cb _xpc_connection_mach_event + 934
18  libdispatch.dylib             	0x00007fff6cf676f8 _dispatch_client_callout4 + 9
19  libdispatch.dylib             	0x00007fff6cf7cbc9 _dispatch_mach_msg_invoke + 435
20  libdispatch.dylib             	0x00007fff6cf6caf6 _dispatch_lane_serial_drain + 263
21  libdispatch.dylib             	0x00007fff6cf7d71c _dispatch_mach_invoke + 481
22  libdispatch.dylib             	0x00007fff6cf6caf6 _dispatch_lane_serial_drain + 263
23  libdispatch.dylib             	0x00007fff6cf6d609 _dispatch_lane_invoke + 414
24  libdispatch.dylib             	0x00007fff6cf76c09 _dispatch_workloop_worker_thread + 596
25  libsystem_pthread.dylib       	0x00007fff6d1c1a3d _pthread_wqthread + 290
26  libsystem_pthread.dylib       	0x00007fff6d1c0b77 start_wqthread + 15

Binary Images:
       0x10a68d000 -        0x10a69afff  com.suprhackersteve.crescendo.CrescendoExtension (1.0.4 - 35) <dec7cffa-8c18-3618-a3f4-4df4da7d0d05> /Library/SystemExtensions/AFA5F0E5-7297-456F-A988-1DE006192220/com.suprhackersteve.crescendo.CrescendoExtension.systemextension/Contents/MacOS/com.suprhackersteve.crescendo.CrescendoExtension
    0x7fff35795000 -     0x7fff35b5afff  com.apple.Foundation (6.9 - 1676.105) <1fa28bab-7296-3a09-8e1e-e62a7d233db8> /System/Library/Frameworks/Foundation.framework/Versions/C/Foundation
    0x7fff35ece000 -     0x7fff35f72ff3  com.apple.framework.IOKit (2.0.2) <a0f54725-036f-3279-a46e-c2abdbfd479b> /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit
    0x7fff69ad3000 -     0x7fff69ae1ffb  libEndpointSecurity.dylib (63.120.3) <be3a6f29-a0f1-3250-94f6-283b97a2b7de> /usr/lib/libEndpointSecurity.dylib
    0x7fff6cf65000 -     0x7fff6cfa5ff0  libdispatch.dylib (1173.100.2) <201edbf3-0b36-31ba-a7cb-443ce35c05d4> /usr/lib/system/libdispatch.dylib
    0x7fff6d101000 -     0x7fff6d12dff7  libsystem_kernel.dylib (6153.121.2) <9f9902c9-a46f-3ca9-b7f9-5ccfe98fbf75> /usr/lib/system/libsystem_kernel.dylib
    0x7fff6d1bf000 -     0x7fff6d1c9fff  libsystem_pthread.dylib (416.100.3) <77488669-19a3-3993-ad65-ca5377e2475a> /usr/lib/system/libsystem_pthread.dylib
    0x7fff6d1f8000 -     0x7fff6d22dffe  libxpc.dylib (1738.120.8) <68d433b6-dcff-385d-8620-f847fb7d4a5a> /usr/lib/system/libxpc.dylib
```

**Comment by SuprHackerSteve** [comment]
Hey @abathur,

Thanks for posting the logs. I see that it's failing when trying to create the ESF client. Is there any endpoint security software running on your host by chance? This is not a typical issue being reported and may need to dig deeper into specifics on your host.

**Comment by abathur** [comment]
I assume the answer is an emphatic _no_ unless you'd apply the categorization to any stock macOS security features. Personal device; 2020 MBA; no third-party security software.

**Comment by SuprHackerSteve** [comment]
Can you run in a VM or other piece of hardware to rule out local issues?

**Comment by abathur** [comment]
I suspect I don't have time to fiddle with setting a macOS VM up (unless it's dead simple and non-interactive--I haven't tried virtualizing macOS), but I can eventually try it on a separate 2018 MBA running 10.15.6 and a 2013 still running some version of Mojave (edit: I guess I'd have to update the latter first, but anyways.)

**Comment by abathur** [comment]
I think I was holding it wrong 🌵 

I tried this on the second system and it worked okay. 

I went through my settings to figure out what the difference was, and for some reason I added the FDE exception for the *app* and not for the *extension*; when I acknowledged the latter and reloaded it appears to be working okay.

It's been enough days that I don't recall exactly how I messed this up, but I vaguely recall adding the FDE exemption before allowing the extension (my preferences may have opened to or even already been open to the Privacy pane, as I've been in there recently troubleshooting something related to developer tools exemptions and syspolicyd assessments). 

I hope I would have had enough firing brain-cells to tick the box next to the extension if it was there, but I'm under that here's-the-hard-deadline-after-missing-several-soft-deadline stress, so who knows. Perhaps I assumed I had to add it manually?
